build_tools/update_environments_and_lock_files.py

```diff
@@ -349,6 +349,9 @@ def remove_from(alist, to_remove):
         # pip-compile, we are going to assume the pip lock file on a Linux
         # 64bit machine gives appropriate versions
         "python_version": "3.8.10",
+        "package_constraints": {
+            "scipy": "1.9.1",  # 1.9.2 not available for 32 bit Windows
```

Comment:
I think pip needs to be configured with the `--prefer-binary` option to prefer the wheel over building the source. Coming from https://github.com/jazzband/pip-tools/pull/1119, pip-tools can be configured to prefer binary with the `PIP_PREFER_BINARY` environment variable.

Given that this is only used to generate the lock files, I slightly prefer the pinning solution that is already in this PR.

Reply:
I agree.

The `PIP_PREFER_BINARY` option is probably a git idea to fix the wheel building problem discussed in https://github.com/scikit-learn/scikit-learn/issues/24612#issuecomment-1273414275.